scripts/test_matrix.py

```diff
@@ -0,0 +1,184 @@
+	['rdm',				prov_rdm],
+	# rdm_rma_simple requires FI_RMA_EVENT
+	['rdm_rma_simple',		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	['rdm_rma_trigger',		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("msg"),			{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("msg","notx"),		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("msg","norx"),		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("rdm"),			{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("rdm","notx"),		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("rdm","norx"),		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("dgram"),		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("dgram","notx"),	{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("dgram","norx"),	{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	['rdm_tagged_peek',		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 1}],
+	['scalable_ep',			{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	['cmatose',			prov_msg],
+	['rdm_shared_av',		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+]
+
+standard = [
+	['msg_pingpong',		prov_msg],
+	['msg_pingpong -v',		prov_msg],
+	['msg_pingpong -k',		prov_msg],
+	['msg_pingpong -k -v',		prov_msg],
+	['msg_bw',			prov_msg],
+	['msg_bw -v',			prov_msg],
+	['msg_bw -k',			prov_msg],
+	['msg_bw -k -v',		prov_msg],
+	['rdm_pingpong',		prov_rdm],
+	['rdm_pingpong -v',		prov_rdm],
+	['rdm_pingpong -k',		prov_rdm],
+	['rdm_pingpong -k -v',		prov_rdm],
+	['rdm_tagged_pingpong',		prov_rdm],
+	['rdm_tagged_pingpong -v',	prov_rdm],
+	['rdm_tagged_pingpong -k',	prov_rdm],
+	['rdm_tagged_pingpong -k -v',	prov_rdm],
+	['rdm_tagged_bw',		prov_rdm],
+	['rdm_tagged_bw -v',		prov_rdm],
+	['rdm_tagged_bw -k',		prov_rdm],
+	['rdm_tagged_bw -k -v',		prov_rdm],
+	['dgram_pingpong',		prov_dgram],
+	['dgram_pingpong -v',		prov_dgram],
+	['dgram_pingpong -k',		prov_dgram],
+	['dgram_pingpong -k -v',	prov_dgram],
+	['rma_bw -e msg -o write',	prov_msg],
+	['rma_bw -e msg -o read',	prov_msg],
+	['rma_bw -e msg -o writedata',	prov_msg],
+	['rma_bw -e rdm -o write',	prov_rdm],
+	['rma_bw -e rdm -o read',	prov_rdm],
+	['rma_bw -e rdm -o writedata',	{'ofi-rxm': 1, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	['msg_rma -o write',		prov_msg],
+	['msg_rma -o read',		prov_msg],
+	['msg_rma -o writedata',	prov_msg],
+	['msg_stream',			prov_msg],
+	['rdm_atomic -o all -I 1000',	{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	['rdm_cntr_pingpong',		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
```

Comment:
rdm_cntr_pingpong doesn't support FI_CONTEXT mode and thus can't run with psm and psm2.